src/apps/search-result/search-result.tsx

```diff
@@ -21,6 +23,36 @@ const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
     0
   );
   const { PagerComponent, page } = usePager(hitcount, pageSize);
+  const [filters, setFilters] = useState<{
+    [key: string]: Record<string, number>;
+  }>({});
+
+  const filterHandler: TagOnclickHandler = ({
+    filterItem: { facet, score, term },
+    action
+  }) => {
+    if (action === "add") {
+      if (Object.keys(filters).includes(facet)) {
+        setFilters({
+          ...filters,
+          [facet]: { ...filters[facet], [term]: score }
+        });
+      } else {
+        setFilters({
+          ...filters,
+          [facet]: { [term]: score }
+        });
+      }
+    }
+
+    if (action === "remove") {
+      if (Object.keys(filters).includes(facet)) {
+        const copy = { ...filters };
+        delete copy[facet][term];
```

Comment:
Yes, of course, we need this so we are not manipulating filters but a clone of filters :)